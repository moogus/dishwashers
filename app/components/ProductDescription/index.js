import React from 'react';

import useGetProduct from 'hooks/product/get-product';

const ProductDescription = () => {
  const { productInformation } = useGetProduct();
  const createMarkup = () => ({ __html: productInformation });

  return (
    <div className="col">
      <span dangerouslySetInnerHTML={createMarkup()} />
    </div>
  );
};

export default ProductDescription;
