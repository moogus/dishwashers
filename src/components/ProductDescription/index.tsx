import React from "react";

import useGetProduct from "../../hooks/product/get-product";

export const ProductDescription = () => {
  const { productInformation } = useGetProduct();
  const createMarkup = () => ({ __html: productInformation });

  return (
    <div className="col">
      <span dangerouslySetInnerHTML={createMarkup()} />
    </div>
  );
};
