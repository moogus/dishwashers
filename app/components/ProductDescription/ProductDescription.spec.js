import React from 'react';
import { render } from '@testing-library/react';

import ProductDescription from 'components/ProductDescription';
import useGetProduct from 'hooks/product/get-product';

jest.mock('hooks/product/get-product');

describe('<ProductDescription />', () => {
  const renderContainer = ({ productId = 'foo' } = {}) => {
    const props = { productId };
    return render(<ProductDescription {...props} />);
  };

  describe('when ProductDescription is rendered', () => {
    beforeEach(() => {
      jest.clearAllMocks();
      useGetProduct.mockReturnValue({ productInformation: '<p>OMG!!!!<p>' });
    });

    test('html string is rendered in the component', () => {
      const { container } = renderContainer();
      expect(container).toMatchSnapshot();
    });
  });
});
