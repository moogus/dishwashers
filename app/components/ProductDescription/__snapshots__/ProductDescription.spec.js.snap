// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductDescription /> when ProductDescription is rendered html string is rendered in the component 1`] = `
<div>
  <div
    class="col"
  >
    <span>
      <p>
        OMG!!!!
      </p>
      <p />
    </span>
  </div>
</div>
`;
